distribution/docker/src/docker/bin/docker-entrypoint.sh

```diff
@@ -38,6 +38,40 @@ if [[ "$1" != "eswrapper" ]]; then
   fi
 fi
 
+# Allow environment variables to be set by creating a file with the
+# contents, and setting an environment variable with the suffix _FILE to
+# point to it. This can be used to provide secrets to a container, without
+# the values being specified explicitly when running the container.
+for VAR_NAME_FILE in $(env | cut -f1 -d= | grep '_FILE$'); do
+  if [[ -n "$VAR_NAME_FILE" ]]; then
+    VAR_NAME="${VAR_NAME_FILE%_FILE}"
+
+    if [[ -n "${!VAR_NAME}" ]]; then
+      echo "ERROR: Both $VAR_NAME_FILE and $VAR_NAME are set. These are mutually exclusive." >&2
```

Comment:
As this is now, this conditional will be skipped if an ENV VAR (say ...`ELASTIC_PASSWORD`) is set to empty content (by accident, for example).

Our code later on will still present this empty env var as `-E ...` cli arg.

What if instead of checking the length of the content of `VAR_NAME` we just ensure it's defined as an env var?

This can be achieved with `env | cut -f1 -d= | grep ^$VAR_NAME\$ >/dev/null` similar to what you've done earlier, or just with `printenv $VAR_NAME >/dev/null`. Both will fail if `VAR_NAME` doesn't match .

Reply:
https://github.com/elastic/elasticsearch/pull/47573/commits/0a07f70223ecd4c8e8061638277f2abdb52c75c0 that resolves this comment looks good, thanks.